ui/src/components/ClubsCarousel/index.tsx

```diff
@@ -17,15 +18,9 @@ export const CLUB_OPTIONS = [
 
 Object.freeze(CLUB_OPTIONS);
 
-interface ClubsCarouselData {
-  name: string;
-  logo: string;
-  category: string;
-}
-
 export const ClubsCarousel: React.FC = () => {
   const [selectedOption, setSelectedOption] = useState(CLUB_OPTIONS[0]);
-  const {data, error} = useAuthFetch('/clubs');
+  const {data, error} = useAuthFetch<PaginatedResponse<ClubData[]>>('/clubs');
```

Comment:
The reason why it should be `PaginatedResponse<ClubData>` instead of `PaginatedResponse<ClubData[]>` is we know paginated response will return a list for sure and the goal of generics is to hide generic features. The goal and requirement aligns.